cmd/commands/datasetIngestor.go

```diff
@@ -259,123 +267,133 @@ For Windows you need instead to specify -user username:password on the command l
+				if err != nil {
+					log.Fatal("Couldn't ingest dataset:", err)
+				}
+				log.Println("Dataset created:", datasetId)
+				// add attachment optionally
+				if addAttachment != "" {
+					log.Println("Adding attachment...")
+					err := datasetIngestor.AddAttachment(client, APIServer, datasetId, metaDataMap, user["accessToken"], addAttachment, addCaption)
 					if err != nil {
-						log.Fatalf("ingestion returned an error: %v\n", err)
+						log.Println("Couldn't add attachment:", err)
 					}
-					// add attachment optionally
-					if addAttachment != "" {
-						err := datasetIngestor.AddAttachment(client, APIServer, datasetId, metaDataMap, user["accessToken"], addAttachment, addCaption)
+					log.Printf("Attachment file %v added to dataset  %v\n", addAttachment, datasetId)
+				}
+				if copyFlag {
+					// TODO rewrite SyncDataToFileserver
+					log.Println("Syncing files to cache server...")
+					err := datasetIngestor.SyncLocalDataToFileserver(datasetId, user, RSYNCServer, datasetSourceFolder, absFileListing, os.Stdout)
+					if err == nil {
+						// delayed enabling
+						archivable = true
+						err := datasetIngestor.MarkFilesReady(client, APIServer, datasetId, user)
 						if err != nil {
-							log.Println("Couldn't add attachment:", err)
-						}
-						log.Printf("Attachment file %v added to dataset  %v\n", addAttachment, datasetId)
-					}
-					if copyFlag {
-						err := datasetIngestor.SyncLocalDataToFileserver(datasetId, user, RSYNCServer, datasetSourceFolder, absFileListing, os.Stdout)
-						if err == nil {
-							// delayed enabling
-							archivable = true
-							datasetIngestor.MarkFilesReady(client, APIServer, datasetId, user)
-						} else {
-							color.Set(color.FgRed)
-							log.Printf("The  command to copy files exited with error %v \n", err)
-							log.Printf("The dataset %v is not yet in an archivable state\n", datasetId)
-							// TODO let user decide to delete dataset entry
-							// datasetIngestor.DeleteDatasetEntry(client, APIServer, datasetId, user["accessToken"])
-							color.Unset()
+							log.Fatal("Couldn't mark files ready:", err)
 						}
+					} else {
+						color.Set(color.FgRed)
+						log.Printf("The  command to copy files exited with error %v \n", err)
+						log.Printf("The dataset %v is not yet in an archivable state\n", datasetId)
+						// TODO let user decide to delete dataset entry
+						// datasetIngestor.DeleteDatasetEntry(client, APIServer, datasetId, user["accessToken"])
+						color.Unset()
 					}
-
-					if archivable {
-						datasetList = append(datasetList, datasetId)
-					}
+					log.Println("Syncing files - DONE")
 				}
-				datasetIngestor.ResetUpdatedMetaData(originalMap, metaDataMap)
 
+				if archivable {
+					archivableDatasetList = append(archivableDatasetList, datasetId)
+				}
 			}
+			datasetIngestor.ResetUpdatedMetaData(originalMap, metaDataMap) // I don't really get this...
```

Comment:
I kinda understood it since then. The whole thing is in a loop that iterates over source directories, and it resets the values of the metadata object to the ones read out from the original file. Of course, the function name does not reflect what it does in effect (a deep copy from one map to another).